### ISSUE TITLE
Invalid routes for routing-api on route_registrar configuration

### ISSUE THREAD

**Issue by JamesClonk** [cloudfoundry/cf-deployment#1055]
### What is this issue about?

routing-api route-registration registers routes that point to not existing host:ports where nothing is listening.

Between `v24.4.0` to `v24.6.0` the PR https://github.com/cloudfoundry/cf-deployment/pull/1014 added the routing-api directly to route-registrar. But as it does not contain or specifies a health-check for route registration, it registers VMs/instances where the routing-api is not actively running, or rather not actively bound to and listening on ports 3000/3001. 

From our observation the routing-api component seems to decide via locket which instance is the master/active and then only that instance of routing-api binds and listens on port 3001, all the other ones do not. Unfortunately this means that if we have 3 API instances, 2 of them will register _api.system-domain/routing_ with tls_port 3001 even though its not actually listening on those instances. This leads to errors when using the routing-api, most easily observed via listing router-groups.

For example CF-CLI/API calls will sporadically fail, show erratic behaviour, with an error message similar to this:
```
$ CF_TRACE=true cf curl /routing/v1/router_groups

REQUEST: [2023-02-22T06:41:23+01:00]
GET /routing/v1/router_groups HTTP/1.1
Host: api.***system-domain***
Accept: application/json
Authorization: [PRIVATE DATA HIDDEN]
Content-Type: application/json
User-Agent: go-cli 7.3.0+645c3ce6a.2021-08-16 / linux

RESPONSE: [2023-02-22T06:41:23+01:00]
HTTP/1.1 502 Bad Gateway
Connection: close
Content-Length: 67
Content-Type: text/plain; charset=utf-8
Date: Wed, 22 Feb 2023 05:41:23 GMT
X-Cf-Routererror: endpoint_failure (dial tcp 100.106.193.27:3001: connect: connection refused)
X-Content-Type-Options: nosniff

502 Bad Gateway: Registered endpoint failed to handle the request.
```
In the above example `100.106.193.27:3001` would be one of the API VMs where routing-api is not actively listening / is not the master.

Asking the Gorouter for its routing table we can see that all API VMs are registered, even though only one them is actively listening on port 3001:
```
$ /var/vcap/jobs/gorouter/bin/retrieve-local-routes
...
  "api.<system-domain>/routing": [
    {
      "address": "100.106.193.14:3001",
      "protocol": "http1",
      "tls": true,
      "ttl": 120,
      "tags": null,
      "private_instance_id": "ecebb846-70d6-4d72-54e4-0fc60c13b736",
      "server_cert_domain_san": "routing-api.service.cf.internal"
    },
    {
      "address": "100.106.193.27:3001",
      "protocol": "http1",
      "tls": true,
      "ttl": 120,
      "tags": null,
      "private_instance_id": "7ba8d120-f8ed-4043-7b15-44524e13da9a",
      "server_cert_domain_san": "routing-api.service.cf.internal"
    },
    {
      "address": "100.106.193.25:3001",
      "protocol": "http1",
      "tls": true,
      "ttl": 120,
      "tags": null,
      "private_instance_id": "f7642dd7-a5ff-4feb-6e68-6310628eddb9",
      "server_cert_domain_san": "routing-api.service.cf.internal"
    }
  ],
...
```

### What version of [cf-deployment](https://github.com/cloudfoundry/cf-deployment/releases) are you using?

cf-deployment v24.4.0 <- works
cf-deployment v24.6.0 or newer <- routing-api fails randomly

### Possible fix? ###

We are currently doing a rather hacky workaround by manually adding a health-check with netcat to the route registration, such as this:
```
...
      route_registrar:
        routes:
        ...
        - name: routing-api
          tls_port: 3001
          server_cert_domain_san: routing-api.service.cf.internal
          registration_interval: 20s
          uris:
          - api.((system_domain))/routing
          health_check:
            name: routing-api-health-check
            script_path: "nc -z 127.0.0.1 3001"
            timeout: 10s
...
```

Something along those lines should probably be implemented for cf-deployment, possibly also by adding a dedicated health-check script for route_registrar to routing-release's routing-api job.

**Comment by ctlong** [comment]
I started a [conversation](https://cloudfoundry.slack.com/archives/C01ABMVNE9E/p1679949435094629) about this in the cf-for-vms-networking channel in slack in order to get the folks who made this change involved.

**Comment by mariash** [comment]
Hi @JamesClonk thank you for reporting this issue. The bug passed though our CI because gorouter is very fast in removing bad backends and retrying 3 times, we were able to reproduce this with 3+ routing API instances. We just pushed a fix to routing-release. It will be released in v0.261.0. And will require a change in cf-deployment - https://github.com/cloudfoundry/cf-deployment/pull/1065
